### PR TITLE
Upgrade rn 0.72.4

### PR DESCRIPTION
**Pull request by coryzoon**
*** Note: The biggest changes made to the code base from the upgrade were:
    1) Replaced react-native-router-flux. This was the library that was used for navigation, it was a wrapper around an older version of React Navigation. react-native-router-flux is no longer maintained and is recommended to replace with React Navigation. A lot of code changes were making updates to the navigation syntax.
    2) Replaced react-native-simple-store with AsyncStorage. Same thing as before, the react-native-simple-store library was a wrapper over the old deprecated AsyncStorage library. I've updated to use the newer version and updated the syntax in the code.
    3) Replaced self-hosted forked react-native-deck-swiper with a newer version of the library. The forked repo was modified to track the panX for the overlay. The newer version of the library can handle this without modifications to it.
    4) ios & android files that are generated by React Native will be different. These files will be rarely, if ever modified (exceptions are noted) and part of the initialization.

Description of files changed:
- .bundle/config - config for the metro bundle, the default bundler for RN.
- .eslintrc/.eslintrc.js - updated to .eslintrc.js and carried over options. It now catches undefined variables (didn't before). 
- .gitignore - update to include new files/directories in newer RN versions.
- Gemfile/Gemfile.lock - iOS cocoapods dependency management.
- README.md - updated to reflected upgrade changes.
- android/* - These are the android files generated by RN and can mostly be ignored in PR since they are generated by RN and not modified. The exceptions would be AndroidManifest.xml, strings.xml, styles.xml, app/build.gradle, & build.gradle. These exceptions were generated by the upgrade and then old options were copied over.
- babel.config.js - modified per React Navigation instructions.
- index.js - modified per React Navigation instructions. Loads Reactotron dev tool when in dev mode.
- ios/* - Same as android. These are the iOS files generated by RN and can mostly be ignored since they are generated by RN and not modified. The exceptions are LaunchScreen.storyboard, Podfile, Podfile.lock.
- jest.config.js - config file generated by RN for jest.
- metro.config.js - config file for metro, the RN bundler, generated by RN.
- package.json, package-lock.json - all dependencies were upgraded to the most recent version except ramda which had breaking changes in the most recent version.
- src/actions/auth.js - updated with new navigation syntax.
- src/actions/classifier.js - updated with new navigation syntax.
- src/actions.index.js - removed functions that were no longer being used and were referencing the deprecated react-native-simple-store.
- src/actions/session.js - updated with AsyncStorage syntax instead of deprecated react-native-simple-store.
- src/components/Discipline.js - updated with new navigation syntax.
- src/components/FullScreenMedial.js - updated "top" property because the close button was out of reach.
- src/components/DrawingClassifier.js - updated with new navigation syntax.
- src/components/MarkableImage.js - fixed bug with unspecified image dimensions.
- src/components/NavBar.js - updated with new navigation drawer syntax.
- src/component/ProjectDisciplines.js - updated to functional component to use the route hook which has access to params. Added new refresh param to handle refreshing the home screen. This replaces the old way of refreshing which was tied to the old navigation system. Also formatted code.
- src/components/Register.js - pass the navigation prop.
- src/components/SideDrawer.js - remove references to the old navigation system. This component is not used anymore but leaving for now to keep PR differences minimal.
- src/components/SideDrawerContent.js - updated with new navigation drawer syntax.
- src/components/SignIn.js - Remove references to the panX that was a part of the old Swiper library. Remove overlay code. This is all handled directly by the react-native-deck-swiper library now.
- src/components/ZooWebView.js - updated with new navigation syntax.
- src/components/classifier/MultiAnswerClassifier - updated with new navigation syntax.
- src/components/classifier/QuestionClassifier - updated with new navigation syntax.
- src/components/classifier/SwipeCard - updated with new navigation syntax.
- src/components/classifier/SwipeClassifier - updated with new navigation syntax. Replaced Swiper component that was a self-hosted forked repo with the updated version of the library. Updated the way it handles background overlay and text where as before it was reliant on the panX event handler from the forked repo, now it is more reliant on the library itself.
- src/components/classifier/SwipeableSubject.js - removed the Dimensions change event handlers since the devices cannot rotate and there were issues with the handler functions. Surrounded the swiping subjects with TouchableWithoutFeed component to allow scrolling down and not have it conflict with the swiping functionality. Fixed bug where on android when you use borderRadius with image: 'contain' it gives an unintended red background.
- src/components/projects/ProjectList.js - updated with new navigation syntax.
- src/components/projects/ProjectTile.js - updated with new navigation syntax.
- src/containers/app.js - replaced react-native-router-flux navigation with RootNavigator from react-navigation. Update to use AsyncStorage for persisting redux. Update to hide splash screen for both iOS/Android.
- src/containers/zooniverseApp.js - updated how isConnected is retrieved. Commented out NotificationModal  which had errors and also didn't appear to be used.
- src/navigation/RootNavigator.js - defines all navigation routes and drawer.
- src/navigators/classiferNavigator.js - updated to pass navigation prop.
- tsconfig.json - standard ts config file that comes with newer RN versions.

### CODE REVIEW

**Comment by lcjohnso** [comment]
@coryzoon FYI re: https://github.com/zooniverse/mobile/pull/458/commits/0b0e2d75656cfa4ab1f675aad5fa0734e0ced732

Totally OK to bump WebView library, but see #357 -- there shouldn't be any need to keep WebView anymore, as we don't support browser-based classification from within the app.

**Comment by lcjohnso** [comment]
@coryzoon -- Thanks for the details re: WebView and beta feedback button, that's very helpful. Sorry for sidetracking PR re: this specific topic.  Totally OK to move this conversation to #357 or elsewhere if you wish to defer this side conversation.

Successful use of a "Provide Beta Feedback" button would be to point to the project-specific form that is manually created when we send out the project for testing.  That link is created and distributed in the beta invite email and posted in the project announcement banner.  There's no automated way of getting this link, so I see no way to make the current button in the app work.

Please remove the button for now -- we can discuss with a broader group possible solutions, but none of them would use the feature as currently implemented:
1. remove the beta group from the app all together
    - not sure we have ever specifically solicited feedback on projects in mobile
2. add functionality to show announcement banner in the app (where correct feedback link is provided)
    - in this case, consider opening link in browser on device, not in browser window within app, so we can drop support for that unused functionality

**Review by mcbouslog — COMMENTED** [review]
Not sure if the related code suggestions are necessary or if `path` should be changed to something else, but the edits were necessary for me to build for iOS.

File: package.json [hunk 1]
```diff
@@ -3,7 +3,6 @@
   "version": "0.0.1",
   "private": true,
   "scripts": {
-    "postinstall": "sed -i -- 's/class Icon extends PureComponent {/class Icon extends PureComponent {constructor(props) { super(props) }/g' node_modules/react-native-vector-icons/lib/create-icon-set.js && patch-package && sed -i '' 's/ArrayList<HashMap<String, String>>\\[\\] packages = this.reactNativeModules/ArrayList<HashMap<String, String>> packages = this.reactNativeModules/g' node_modules/@react-native-community/cli-platform-android/native_modules.gradle",
     "start": "node node_modules/react-native/local-cli/cli.js start",
```
Comment (mcbouslog) [inline package.json:6]: Is this common? I expected just `react-native start` here. Not sure it makes any difference, but can't hurt to note.

Comment (coryzoon) [reply package.json:6]: It's not common, I don't think that any of the scripts are actually needed and over time I would probably remove them from the readme but to keep the PR small I copied everything over. The postinstall script was actually causing issues when building and was not needed so I opted to remove it.

File: ios/ZooniverseMobile.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -3,528 +3,96 @@
...
+		09561D872AAF44DE00927AE0 /* zoo-font.ttf */ = {isa = PBXFileReference; lastKnownFileType = file; name = "zoo-font.ttf"; path = "../../mobile-old/android/app/src/main/assets/fonts/zoo-font.ttf"; sourceTree = "<group>"; };
+		09561D882AAF44DE00927AE0 /* Karla-Italic.ttf */ = {isa = PBXFileReference; lastKnownFileType = file; path = "Karla-Italic.ttf"; sourceTree = "<group>"; };
+		09561D892AAF44DE00927AE0 /* Karla-Bold.ttf */ = {isa = PBXFileReference; lastKnownFileType = file; path = "Karla-Bold.ttf"; sourceTree = "<group>"; };
+		09561D8A2AAF44DF00927AE0 /* Karla-BoldItalic.ttf */ = {isa = PBXFileReference; lastKnownFileType = file; path = "Karla-BoldItalic.ttf"; sourceTree = "<group>"; };
+		09561D8B2AAF44DF00927AE0 /* OpenSans-Bold.ttf */ = {isa = PBXFileReference; lastKnownFileType = file; name = "OpenSans-Bold.ttf"; path = "../../mobile-old/android/app/src/main/assets/fonts/OpenSans-Bold.ttf"; sourceTree = "<group>"; };
+		09561D8C2AAF44DF00927AE0 /* OpenSans-Regular.ttf */ = {isa = PBXFileReference; lastKnownFileType = file; name = "OpenSans-Regular.ttf"; path = "../../mobile-old/android/app/src/main/assets/fonts/OpenSans-Regular.ttf"; sourceTree = "<group>"; };
```
[the remark this answers is not in the document]
Comment (coryzoon) [reply ios/ZooniverseMobile.xcodeproj/project.pbxproj:42]: Thank you Mark, the "mobile-old" directory was a copy of the mobile app before the upgrades so I could keep it as a reference. Maybe something accidentally got copied over. I'll commit these suggestions.

File: src/components/classifier/QuestionClassifier.js [hunk 3]
```diff
@@ -158,12 +156,12 @@ class QuestionClassifier extends Component {
         const seenThisSession = R.indexOf(subject.id, subjectsSeenThisSession) >= 0
         const classificationPanel =
             <View
-                style={[styles.classificationPanel, colorModes.framingBackgroundColorFor(this.props.project.in_museum_mode)]}>
+                style={[styles.classificationPanel, colorModes.framingBackgroundColorFor(this.props.route.params.project.route.params.in_museum_mode)]}>
```
Comment (mcbouslog) [inline src/components/classifier/QuestionClassifier.js:159]:
```suggestion
                style={[styles.classificationPanel, colorModes.framingBackgroundColorFor(this.props.route.params.project.in_museum_mode)]}>
```

File: src/components/classifier/QuestionClassifier.js [hunk 4]
```diff
@@ -98,15 +98,13 @@ class QuestionClassifier extends Component {
...
-            project,
-            workflow,
             subject,
             subjectsSeenThisSession,
             task,
-            inBetaMode,
             guide,
             answers
         } = this.props
+        const { project, workflow, inBetaMode } = this.props.route.params;
```
Comment (mcbouslog) [inline src/components/classifier/QuestionClassifier.js:107]: Similar to this line, I think line 73 within `submitClassification` could use `const { workflow } = this.props.route.params;` and workflow can be removed from the props destructuring on line 79.

File: package.json [hunk 5]
```diff
@@ -3,7 +3,6 @@
...
     "android": "react-native run-android --appIdSuffix 'debug'",
```
Comment (mcbouslog) [inline package.json:7]:
```suggestion
    "android": "react-native run-android",
```

**Review by mcbouslog — APPROVED** [review]
This is looking great! 🎉 

I'm going to approve this, but note some issues in comments and as follows. I'm thinking maybe commit suggestions and if any fixes small and straightforward they can be added to this PR, otherwise we can fix in subsequent PRs?

I think to look into for subsequent PRs:
- 🐛 if select "mobile drawing project" project, I can't click back to projects list or get out of classifying, I'm not sure if it's the project name or description or something else
- 🐛 if select "Mobile Testing - Staging", workflow "MultiAnswer MultiFrames" can't submit a classification
- are the pngs in `ios/ZooniverseMobile/Images.xcassets/fulllaunch.imageset` duplicates?

**Comment by mcbouslog** [comment]
@coryzoon - [comment](https://github.com/zooniverse/mobile/pull/458#issuecomment-1748798706) LGTM, nothing missing I can think of! Let's get this merged and we'll go from there 🚀 🎉

**Pull request merged**